azure-iot-device/tests/common/pipeline/test_pipeline_flow.py

```diff
@@ -0,0 +1,216 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import logging
+import pytest
+from azure.iot.device.common.pipeline import (
+    pipeline_thread,
+    pipeline_stages_base,
+    pipeline_ops_base,
+    pipeline_events_base,
+    pipeline_exceptions,
+)
+from azure.iot.device.common import handle_exceptions
+from tests.common.pipeline.helpers import (
+    make_mock_stage,
+    assert_callback_failed,
+    assert_callback_succeeded,
+)
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+# This fixture makes it look like all test in this file  tests are running
+# inside the pipeline thread.  Because this is an autouse fixture, we
+# manually add it to the individual test.py files that need it.  If,
+# instead, we had added it to some conftest.py, it would be applied to
+# every tests in every file and we don't want that.
+@pytest.fixture(autouse=True)
+def apply_fake_pipeline_thread(fake_pipeline_thread):
+    pass
+
+
+class MockPipelineStage(pipeline_stages_base.PipelineStage):
+    def _execute_op(self, op):
+        self._send_op_down(op)
+
+
+@pytest.fixture
+def stage(mocker, arbitrary_exception, arbitrary_base_exception):
+    return make_mock_stage(
+        mocker=mocker,
+        stage_to_make=MockPipelineStage,
+        exc_to_raise=arbitrary_exception,
+        base_exc_to_raise=arbitrary_base_exception,
+    )
+
+
+@pytest.mark.describe("PipelineFlow._send_worker_op_down()")
```

Comment:
>PipelineFlow._send_worker_op_down() [](start = 23, length = 35)

The unit syntax we defined would correctly be

`PipelineFlow - ._send_worker_op_down()`